### DISCUSSION TITLE
umdp3_checker rose-stem test failing

### DISCUSSION THREAD

**Discussion by johnrostron** [MetOffice/simulation-systems#115] [UM]
Hi,
I'm running rose-stem on a vn13.4 test branch, and everything seems to be working except for 'script_spice_umdp3_checker'. This fails with the following error message:

/usr/bin/env: perl
: No such file or directory
[FAIL] module_wrapper $CYLC_SUITE_SHARE_DIR/imported_github_scripts/script_umdp3_checker/bin/umdp3_check.pl $SOURCE_UM $CYLC_SUITE_SHARE_DIR/imported_github_scripts/script_umdp3_checker/file/whitelist_includes.txt <<'__STDIN__'
[FAIL] 
[FAIL] '__STDIN__' # return-code=127

Has anyone come across this error before, and could anyone advise on what might be going wrong?

Thanks!
John

**Comment by richardgilham** [comment]
Hi John,

Did you try retriggering? It could be a transient disk problem.

**Reply by johnrostron** [reply]
Hi Richard,

I have tried retriggering the task, and re-running the test suite, but with the same result.

**Reply by johnrostron** [reply]
I can get the task to run if I change the command from 'module_wrapper $CYLC_SUITE_SHARE_DIR/imported_github_scripts...' to 'module_wrapper **perl** $CYLC_SUITE_SHARE_DIR/imported_github_scripts...'.

So it looks like there's some problem with running that script as an executable. The permissions look ok ('-rwxr-xr-x'). Any ideas what might be going wrong?

**Reply by richardgilham** [reply]
Curious- we don't see any problems running that test. Anything novel/interesting in your .profile or .bashrc etc?

FWIW, I wouldn't worry about this too much for a test branch. If the test runs OK in your dev branch, that'll be fine for review purposes.

Rich

**Reply by johnrostron** [reply]
I tried temporarily removing my .profile and .bashrc etc, but I still got the same error

**Reply by johnrostron** [reply]
I think I've made progress on this. [This stackoverflow post](https://stackoverflow.com/questions/34791031/usr-bin-env-perl-does-work-as-command-but-not-in-perl-script) described my problem, which seems to be because the perl script (umdp3_check.pl) has 'DOS line endings'. I can resolve the error by running 'dos2unix' on the script held in the cylc-run directory. But it seems the script downloaded by 'script_spice_export_git_scripts' always has the DOS line endings. Is there some way to correct this, without me having to run 'dos2unix' manually?

**Reply by richardgilham** [reply]
Can you point us to your branch(es)/ticket? If we can rerun successfully we mark it down as an oddity and let you crack on with what you really want to achieve.

**Reply by johnrostron** [reply]
Thanks Richard, here are link to the ticket/branches: 

Ticket: https://code.metoffice.gov.uk/trac/um/ticket/7491
Dev branch: https://code.metoffice.gov.uk/trac/um/log/main/branches/dev/johnrostron/vn13.4_GAL9PPE_ConvParams
Test branch: https://code.metoffice.gov.uk/trac/um/log/main/branches/test/johnrostron/r121692_GAL9PPE_ConvParams_test

**Reply by johnrostron** [reply]
I have put the commands:

git config --global core.autocrlf input
git config --global core.safecrlf false

in my .bash_profile and I can now run the umdp3 check successfully without having to manually apply 'dos2unix' to the perl script.
